CMakeLists.txt

```diff
@@ -18,19 +18,77 @@ option(ENABLE_MPI "Build MPI support" OFF)
 option(ENABLE_CUDA "Build CUDA support" OFF)
 option(ENABLE_OPENMP "Build OpenMP support" OFF)
 
+# Advanced configuration options
+# TODO: If these are disabled, the code will not compile or run correctly.
+#       Fix those issues so that these options are actually configurable.
+option(ENABLE_PICK "Enable pick and set methods on ManagedArrays" ON)
+option(ENABLE_PINNED "Enable pinned memory space (used for scans)" ON)
+
 # Extra components
 option(ENABLE_TESTS "Build tests" ON)
 option(ENABLE_BENCHMARKS "Build benchmarks" ON)
 option(ENABLE_DOCUMENTATION "Build documentation" ON)
 option(ENABLE_EXAMPLES "Build examples" ON)
 
-# Dependencies
+# TODO: Not currently used, but allows ENABLE_EXERCISES to be set properly
+#       for submodules. Investigate alternatives.
+option(ENABLE_EXERCISES "Build exercises" OFF)
+
+# Extra submodule components
+option(ENABLE_SUBMODULE_TESTS "Build submodule tests" OFF)
+option(ENABLE_SUBMODULE_BENCHMARKS "Build submodule benchmarks" OFF)
+option(ENABLE_SUBMODULE_DOCUMENTATION "Build submodule documentation" OFF)
+option(ENABLE_SUBMODULE_EXAMPLES "Build submodule examples" OFF)
+option(ENABLE_SUBMODULE_EXERCISES "Build submodule exercises" OFF)
+
+# Required dependencies (by default uses submodules)
 set(BLT_SOURCE_DIR "" CACHE PATH "Path to external BLT")
-set(camp_DIR "" CACHE PATH "Path to external campConfig.cmake or camp-config.cmake")
+
+# TODO: When RAJA is updated past v0.11.0, camp_DIR should point to the
+#       directory containing campConfig.cmake or camp-config.cmake.
+#       Update the variable documentation to reflect that change.
+set(camp_DIR "" CACHE PATH "Path to external CAMP (top level of the install or source directory)")
+
+set(umpire_DIR "" CACHE PATH "Path to external umpireConfig.cmake or umpire-config.cmake")
+set(raja_DIR "" CACHE PATH "Path to external rajaConfig.cmake or raja-config.cmake")
+set(chai_DIR "" CACHE PATH "Path to external chaiConfig.cmake or chai-config.cmake")
```

Comment:
why are these lower case (chai_DIR)? Wouldn't uppercase be more consistent with other options (CHAI_DIR) ?

Reply:
This is actually a cmake thing. If you specify chai_DIR and call find_package(chai), it will look for chaiConfig.cmake or chai-config.cmake in the directory pointed to by chai_DIR. In this situation it is case sensitive.

Reply:
`make install` in CHAI creates chaiConfig.cmake instead of CHAIConfig.cmake, so we need to use the lower case version.

Reply:
Do you think it would make sense to have CHAI_DIR etc for external users to be consistent with how other options and other packages work, and then copy those values to chai_DIR internally?

Reply:
I've made it so we can use CHAI_DIR or chai_DIR.